### PR TITLE
Updated twig requirement

### LINKED ISSUE #522: Missing Twig requirement
_Not up front: I have very little experience with Symfony in general, so my diagnosis may well be wrong._

### Setup

We have a simple project, without any networked interface at the moment, so no template engine is used. The `composer.json` defines the dependency as
```
    "require-dev": {
        "sensio/generator-bundle": "~3.0",
        ...
```
The actually installed version is 3.0.11.

### Problem

Trying to generate a bundle (`app/console generate:bundle --namespace=Fou/BarreApiBundle`) fails because the project. The error is
```
PHP Fatal error:  Class 'Twig_Environment' not found in [...]/vendor/sensio/generator-bundle/Generator/Generator.php on line 50
```
so it seems that Twig is somehow missing.

### Workaround

Run `composer require --dev twig/twig=~1.18` (the version mentioned as require-dev in SensioGeneratorBundle) to install Twig.

### Suggested Fix

The GeneratorBundle itself should require Twig. If Twig is optional (only required for some functionality), it should at least suggest it.

Cheers & happy hacking!

Uli

### PR DESCRIPTION
**Pull request by HeahDude**
Closes #522.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,12 +14,12 @@
         "symfony/framework-bundle": "~2.7|~3.0",
         "symfony/console": "~2.7|~3.0",
         "symfony/process": "~2.7|~3.0",
-        "symfony/yaml": "~2.7|~3.0"
+        "symfony/yaml": "~2.7|~3.0",
+        "twig/twig": "~1.28.1"
```
Comment (xabbuh) [inline composer.json]: should be `^1.28.1`

Comment (xabbuh) [inline composer.json]: And I think you should then also allow Twig 2: `^1.28.1||^2.0`

Comment (HeahDude) [inline composer.json]: Fixed.

**Comment by xabbuh** [comment]
👍

File: composer.json [hunk 2]
```diff
@@ -14,12 +14,12 @@
         "symfony/framework-bundle": "~2.7|~3.0",
         "symfony/console": "~2.7|~3.0",
         "symfony/process": "~2.7|~3.0",
-        "symfony/yaml": "~2.7|~3.0"
+        "symfony/yaml": "~2.7|~3.0",
+        "twig/twig": "^1.28.2||^2.0"
```
Comment (fabpot) [inline composer.json]: What about `^1.28.2|^2.0` to be consistent with the other constraints?

**Comment by HeahDude** [comment]
ping @fabpot

**Comment by fabpot** [comment]
Thank you @HeahDude.

**Pull request merged**